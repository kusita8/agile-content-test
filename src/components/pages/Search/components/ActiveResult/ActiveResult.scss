.activeresult {
  padding: 1.4rem;
  border-radius: 0.4rem;
  border: 0.1rem solid var(--grey-c);
  display: flex;
  flex-direction: column;
  gap: 2rem;
  background-color: white;

  @media only screen and (max-width: 600px) {
    width: 80%;
    margin: 0 auto;
  }

  &__image {
    height: 25rem;
    width: 100%;
    border-radius: 0.2rem;
    overflow: hidden;

    img {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }
}
