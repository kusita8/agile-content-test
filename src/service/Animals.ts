import { faker } from "@faker-js/faker";

const results = [...new Array(100)].map(() => {
  const type = faker.animal.type() as keyof typeof faker.animal;

  return {
    type,
    id: faker.datatype.uuid(),
    url: faker.internet.url(),
    title: faker.animal[type](),
    description: faker.lorem.sentences(),
    image: faker.image.animals(644, 362, true),
  };
});

export const getAnimals = async (params: Record<string, string>) => {
  // fake delay
  await new Promise((res) => setTimeout(res, 1000));

  const { query } = params;
  const lowercaseQuery = query.toLowerCase();

  return results.filter(
    (animal) =>
      lowercaseQuery &&
      (animal.type.toLowerCase().includes(lowercaseQuery) ||
        animal.title.toLowerCase().includes(lowercaseQuery))
  );
};
