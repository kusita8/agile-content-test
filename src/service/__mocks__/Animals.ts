import { faker } from "@faker-js/faker";

export const getAnimal = () => {
  const type = faker.animal.type() as keyof typeof faker.animal;

  return {
    type: faker.animal.type(),
    id: faker.datatype.uuid(),
    url: faker.internet.url(),
    title: faker.animal[type](),
    description: faker.lorem.sentences(),
    image: faker.image.animals(644, 362, true),
  };
};

export const getAnimals = async (params: Record<string, string>) => {
  const { query } = params;

  const type = faker.animal.type() as keyof typeof faker.animal;

  return [...new Array(2)]
    .map(() => ({
      type: faker.animal.type(),
      id: faker.datatype.uuid(),
      url: faker.internet.url(),
      title: faker.animal[type](),
      description: faker.lorem.sentences(),
      image: faker.image.animals(644, 362, true),
    }))
    .filter(
      (animal) =>
        query &&
        animal.type.toLocaleLowerCase().includes(query.toLocaleLowerCase())
    );
};
