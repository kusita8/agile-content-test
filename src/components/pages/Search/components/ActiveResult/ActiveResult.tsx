import { FC } from "react";

import { Result } from "src/components/pages/Search/components/Result/Result";
import { IAnimal } from "src/types/Animal.interface";

import "./ActiveResult.scss";

export const ActiveResult: FC<IAnimal> = (props) => {
  return (
    <div className="activeresult">
      <div className="activeresult__image">
        <img src={props.image} alt={props.title} />
      </div>
      <Result activeDisplay {...props} />
    </div>
  );
};
