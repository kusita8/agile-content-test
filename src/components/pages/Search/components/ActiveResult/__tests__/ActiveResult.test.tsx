/* eslint-disable no-restricted-imports */
import { ActiveResult } from "src/components/pages/Search/components/ActiveResult/ActiveResult";
import { getAnimal } from "src/service/__mocks__/Animals";
import { renderWithProviders } from "src/tests/renderWithProviders";

describe("ActiveResult", () => {
  test("it should match the snapshot", () => {
    const { asFragment } = renderWithProviders(
      <ActiveResult {...getAnimal()} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
