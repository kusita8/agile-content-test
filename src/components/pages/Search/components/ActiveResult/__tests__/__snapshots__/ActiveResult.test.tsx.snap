// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActiveResult > it should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <div
      class="header"
    >
      <p>
        <b>
          Agile content
        </b>
         Frontend test
      </p>
      <div
        class="header__right"
      >
        <svg
          class="gb_p"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6,8c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM12,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM6,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM6,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM12,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM16,6c0,1.1 0.9,2 2,2s2,-0.9 2,-2 -0.9,-2 -2,-2 -2,0.9 -2,2zM12,8c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM18,14c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2zM18,20c1.1,0 2,-0.9 2,-2s-0.9,-2 -2,-2 -2,0.9 -2,2 0.9,2 2,2z"
          />
        </svg>
        <div
          class="header__user"
        >
          <img
            alt="Profile image"
            src="https://images.unsplash.com/photo-1511367461989-f85a21fda167?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2231&q=80"
          />
        </div>
      </div>
    </div>
    <div
      class="activeresult"
    >
      <div
        class="activeresult__image"
      >
        <img
          alt="Standard Schnauzer"
          src="https://loremflickr.com/644/362/animals?lock=66940"
        />
      </div>
      <div
        class="result result--active-display"
      >
        <a
          class="result__link"
          href="http://nutritious-service.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            class="result__url"
          >
            http://nutritious-service.com
          </div>
          <h4>
            Standard Schnauzer
          </h4>
        </a>
        <div
          class="result__description"
        >
          Itaque esse suscipit debitis. Necessitatibus culpa perspiciatis cum doloribus.
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <p>
        Â© Google 2023
      </p>
      <p>
        version: 0.1.0
      </p>
    </div>
  </div>
</DocumentFragment>
`;
